### PR TITLE
Feat: add condition to most audits

### PR DESCRIPTION
**Pull request by z3z1ma**
Most builtin audits can be made further general purpose by adding an optional `@condition` parameter. This keeps existing behavior as-is but allows more flexibility before needing to roll your own 

cc @plaflamme

### CODE REVIEW

**Review by plaflamme — APPROVED** [review]

File: sqlmesh/core/audit/builtin.py [hunk 1]
```diff
@@ -7,22 +7,26 @@
 # not_null(columns=[column_1, column_2])
 not_null_audit = ModelAudit(
     name="not_null",
+    defaults={"condition": exp.true()},
```
Comment (izeigerman) [inline sqlmesh/core/audit/builtin.py:10]: I can't recall whether plain `True` universally works across all engines. I remember having to do `1 = 1` expression instead. Or maybe it's only relevant when there is just one condition in the `WHERE` clause. @tobymao @eakmanrq @treysp any chance one of you knows about what I'm referring to?

Comment (tobymao) [reply sqlmesh/core/audit/builtin.py:10]: why do you need any condition at all? sqlglot will remove empty where statements

Comment (tobymao) [reply sqlmesh/core/audit/builtin.py:10]: i think TRUE doesn't work in tsql, only 1 =1, but i think we may have a sqlglot transpiler for that

Comment (treysp) [reply sqlmesh/core/audit/builtin.py:10]: sqlglot transpiles to 1=1

```
>>> exp.true().sql(dialect="tsql")
'(1 = 1)'
```

Comment (z3z1ma) [reply sqlmesh/core/audit/builtin.py:10]: @tobymao what do you mean, how would a user be able to _optionally_ supply a condition?
If they dont provide one, we need a default in the context so there is no KeyError when trying to render the expression?
If the user does provide a condition, we need to specify where it goes (in the where clause with @condition)

Can you share and example of what you mean?

Comment (tobymao) [reply sqlmesh/core/audit/builtin.py:10]: i guess we need a new macro def and_ or def or_ that will take in optionals and construct the proper query, can chat with you over slack if you need more info

sqlglot has helpers

exp.and_ that does this i think

**Comment by tobymao** [comment]
taking this to the finish line here https://github.com/TobikoData/sqlmesh/pull/2255

**Pull request closed without merging**